cmd/root.go

```diff
@@ -37,37 +38,66 @@ var RootCmd = &cobra.Command{
 Build, start and manage service instances with a single command.`,
 	SilenceUsage: true,
 	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+
+		var dirConfig *home.EdwardConfiguration
+		var err error
+		dirConfig = &home.EdwardConfiguration{}
+		if edwardHome != "" {
+			err = dirConfig.InitializeWithDir(edwardHome)
+		} else {
+			err = dirConfig.Initialize()
+		}
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
 		if redirectLogs {
 			logger = log.New(os.Stdout, fmt.Sprintf("%v >", os.Args), log.Ldate|log.Ltime|log.Lmicroseconds|log.Lshortfile)
 		}
-		if logFile != "" {
-			prefix := "edward"
-			if len(args) > 0 {
-				prefix = fmt.Sprintf("%s %s", cmd.Use, args[0])
-			}
-			logger = log.New(&lumberjack.Logger{
-				Filename:   logFile,
-				MaxSize:    50, // megabytes
-				MaxBackups: 30,
-				MaxAge:     1, //days
-			}, fmt.Sprintf("%s > ", prefix), log.Ldate|log.Ltime|log.Lmicroseconds|log.Lshortfile)
+
+		prefix := "edward"
+		if len(args) > 0 {
+			prefix = fmt.Sprintf("%s %s", cmd.Use, args[0])
 		}
+		logger = log.New(&lumberjack.Logger{
+			Filename:   path.Join(dirConfig.EdwardLogDir, "edward.log"),
+			MaxSize:    50, // megabytes
+			MaxBackups: 30,
+			MaxAge:     1, //days
+		}, fmt.Sprintf("%s > ", prefix), log.Ldate|log.Ltime|log.Lmicroseconds|log.Lshortfile)
+
 		// Begin logging
 		logger.Printf("=== Edward v%v ===\n", common.EdwardVersion)
 		logger.Printf("Args: %v\n", os.Args)
+		// Recover from any panic and log appropriately
+		defer func() {
+			if r := recover(); r != nil {
+				fmt.Println("Recovered from panic:", r)
+				if logger != nil {
```

Comment:
it seems impossible for logger to be nil here?

Reply:
It's unlikely, but logger could have been set to nil after this defer was created, figured it was better safe than sorry.